### PR TITLE
dcos package Install with universe v2

### PR DESCRIPTION
**Pull request by tamarrow**

### CODE REVIEW

File: cli/tests/data/dcos_universe2.toml [hunk 1]
```diff
@@ -0,0 +1,9 @@
+[core]
```
Comment (jsancio) [inline cli/tests/data/dcos_universe2.toml]: @tamarrow Let's not just execute a subset of the tests against version-2 of universe. All of the cli tests should be against version-2.x of universe.

File: dcos/package.py [hunk 2]
```diff
@@ -1331,6 +1331,32 @@ def has_marathon_definition(self, revision):
...
+    def get_marathon_json_file(self, revision):
+        """Returns the file name of Marathon json
+
+        :param revision: package revision
+        :type revision: str
+        :returns: Marathon file name
+        :rtype: str
+        """
+        if self.has_marathon_definition(revision):
+            return 'marathon.json'
```
Comment (jsancio) [inline dcos/package.py]: Let's make it work only with version-2. We don't need this check if so, no?

File: dcos/package.py [hunk 3]
```diff
@@ -1364,8 +1390,11 @@ def marathon_json(self, revision, options):
         :rtype: dict
         """
 
+        package_json = self.package_json(revision)
+        options = _merge_options(options, {"package": package_json})
```
Comment (jsancio) [inline dcos/package.py]: @tamarrow The user should not be able to overwrite stuff in `package.json`. That means that we need to merge user values with the default and then merge package.json with the previous result. Are we doing that? Can we have tests that check this?

Also, I thought we agreed to exit with an error if we overwrite package values.

File: dcos/package.py (hunk 2 above)
Comment (tamarrow) [inline dcos/package.py]: updating

File: dcos/package.py (hunk 3 above)
Comment (tamarrow) [inline dcos/package.py]: yes, will add a test

File: cli/dcoscli/data/universe-schema/version.json [hunk 4]
```diff
@@ -0,0 +1,3 @@
+{
+          "version": "1.0.0-rc1"
+}
```
Comment (jsancio) [inline cli/dcoscli/data/universe-schema/version.json]: What is this file about? Do we use it anywhere?

File: cli/tests/data/package/helloworld-bad.json [hunk 5]
```diff
@@ -0,0 +1,3 @@
+{
+    "port": "hi"
+}
```
Comment (jsancio) [inline cli/tests/data/package/helloworld-bad.json]: What is this about? Do we use it anywhere?

File: cli/tests/data/package/json/cassandra_single_node.json [hunk 6]
```diff
@@ -0,0 +1,12 @@
...
+    "resources": {
+      "cpus": 0.1,
+      "mem": 512,
+      "disk": 272
+    },
+    "health-check-interval-seconds": 15,
+    "node-count": 1,
+    "seed-count": 1
+  }
+}
```
Comment (jsancio) [inline cli/tests/data/package/json/cassandra_single_node.json]: Same here. Do we use it anywhere?

File: cli/tests/data/package/json/test_describe_app_cassandra.json [hunk 7]
```diff
@@ -0,0 +1,69 @@
...
+,"CASSANDRA_RESOURCE_HEAP_MB": "{{cassandra.resource.heap-mb}}"
+{{/cassandra.resource.heap-mb}}
+{{#cassandra.framework.authentication.principal}}
+    ,"DEFAULT_PRINCIPAL": "{{cassandra.framework.authentication.principal}}"
+{{/cassandra.framework.authentication.principal}}
+{{#cassandra.framework.authentication.secret}}
+    ,"DEFAULT_SECRET": "{{cassandra.framework.authentication.secret}}"
+{{/cassandra.framework.authentication.secret}}
+  }
+}
```
Comment (jsancio) [inline cli/tests/data/package/json/test_describe_app_cassandra.json]: Same here. Do we use this anywhere?

Comment (tamarrow) [inline cli/tests/data/package/json/test_describe_app_cassandra.json]: just removed a bunch of unused files.

File: cli/tests/data/package/json/test_describe_cassandra.json [hunk 8]
```diff
@@ -0,0 +1,33 @@
...
+  "postInstallNotes": "Thank you for installing the Apache Cassandra DCOS Service.\n\n\tDocumentation: http://mesosphere.github.io/cassandra-mesos/\n\tIssues: https://github.com/mesosphere/cassandra-mesos/issues",
+  "postUninstallNotes": "The Apache Cassandra DCOS Service has been uninstalled and will no longer run.\nPlease follow the instructions at http://docs.mesosphere.com/services/cassandra/#uninstall to clean up any persisted state",
+  "preInstallNotes": "The Apache Cassandra DCOS Service implementation is alpha and there may be bugs, incomplete features, incorrect documentation or other discrepancies.\nThe default configuration requires 3 nodes each with 0.3 CPU shares, 1184MB of memory and 272MB of disk.",
+  "scm": "https://github.com/mesosphere/cassandra-mesos.git",
+  "tags": [
+    "mesosphere",
+    "framework"
+  ],
+  "version": "0.1.0-1"
+}
```
Comment (jsancio) [inline cli/tests/data/package/json/test_describe_cassandra.json]: Same here. Do we use this anywhere?

File: cli/tests/data/package/json/test_describe_cassandra_package_version.json [hunk 9]
```diff
@@ -0,0 +1,33 @@
...
+  "postInstallNotes": "Thank you for installing the Apache Cassandra DCOS Service.\n\n\tDocumentation: http://mesosphere.github.io/cassandra-mesos/\n\tIssues: https://github.com/mesosphere/cassandra-mesos/issues",
+  "postUninstallNotes": "The Apache Cassandra DCOS Service has been uninstalled and will no longer run.\nPlease follow the instructions at http://docs.mesosphere.com/services/cassandra/#uninstall to clean up any persisted state",
+  "preInstallNotes": "The Apache Cassandra DCOS Service implementation is alpha and there may be bugs, incomplete features, incorrect documentation or other discrepancies.\nThe default configuration requires 3 nodes each with 0.3 CPU shares, 1184MB of memory and 272MB of disk.",
+  "scm": "https://github.com/mesosphere/cassandra-mesos.git",
+  "tags": [
+    "mesosphere",
+    "framework"
+  ],
+  "version": "0.1.0-1"
+}
```
Comment (jsancio) [inline cli/tests/data/package/json/test_describe_cassandra_package_version.json]: Same here. Do we use this anywhere?

File: cli/tests/data/package/json/test_describe_options.json [hunk 10]
```diff
@@ -0,0 +1,62 @@
...
+  },
+  "mem": 512,
+  "ports": [
+    0
+  ],
+  "uris": [
+    "https://raw.githubusercontent.com/mesosphere/universe/cli-test-3/repo/packages/C/cassandra/0/assets/cassandra-mesos-0.1.0-1.tar.gz",
+    "https://raw.githubusercontent.com/mesosphere/universe/cli-test-3/repo/packages/C/cassandra/0/assets/jre-7u76-linux-x64.tar.gz"
+  ]
+}
```
Comment (jsancio) [inline cli/tests/data/package/json/test_describe_options.json]: Same here. Do we use this anywhere?

File: cli/tests/integrations/common.py [hunk 11]
```diff
@@ -597,3 +604,37 @@ def config_unset(key, index=None, env=None):
 
     assert returncode == 0
     assert stderr == b''
+
+
+@contextlib.contextmanager
+def _package(name,
```
Comment (jsancio) [inline cli/tests/integrations/common.py]: Used outside the package, no? Let's remove the `_` and rename to `_package`.

Comment (jsancio) [inline cli/tests/integrations/common.py]: Actually, I couldn't find a usage for this in this PR.

File: cli/tests/integrations/common.py [hunk 12]
```diff
@@ -597,3 +604,37 @@ def config_unset(key, index=None, env=None):
...
+                   stdout=stdout)
+    try:
+        yield
+    finally:
+        assert_command(
+            ['dcos', 'package', 'uninstall', name],
+            stderr=uninstall_stderr)
+
+
+def list_packages(args=['--json'], stdout=b'[]\n'):
```
Comment (jsancio) [inline cli/tests/integrations/common.py]: I couldn't find an usage for this method.

File: dcos/package.py [hunk 13]
```diff
@@ -1321,6 +1328,16 @@ def has_command_definition(self, revision):
 
         return self.has_definition(revision, 'command.json')
 
+    def has_resource_definition(self, revision):
```
Comment (jsancio) [inline dcos/package.py]: We added a few new methods to this file. Can we make them private if they are not used outside of this class? It looked like they are all used in the `marathon_json` method.

**Comment by tamarrow** [comment]
@jsancio updated

File: dcos/package.py (hunk 13 above)
Comment (tamarrow) [inline dcos/package.py]: done

**Comment by jsancio** [comment]
LGTM. :shipit: :confetti_ball:

**Pull request merged**